clang-tools-extra/clang-tidy/ClangTidyDiagnosticConsumer.cpp

```diff
@@ -311,7 +311,12 @@ ClangTidyDiagnosticConsumer::ClangTidyDiagnosticConsumer(
     : Context(Ctx), ExternalDiagEngine(ExternalDiagEngine),
       RemoveIncompatibleErrors(RemoveIncompatibleErrors),
       GetFixesFromNotes(GetFixesFromNotes),
-      EnableNolintBlocks(EnableNolintBlocks) {}
+      EnableNolintBlocks(EnableNolintBlocks) {
+
+  if (Context.getOptions().ExcludeHeaderFilterRegex)
+    ExcludeHeaderFilter = std::make_unique<llvm::Regex>(
+        *Context.getOptions().ExcludeHeaderFilterRegex);
```

Comment:
i thing this doesn't work, i mean that ExcludeHeaderFilterRegex is always set because init value is set to '' and is used as default value for option, therfor ExcludeHeaderFilterRegex will never be nullptr. try adding check here to verify also that string is not empty.

Reply:
What makes you suspect it doesn't work? I did some testing by placing `abort()` in an else clause and it caused many unit tests to crash. That leads me to believe it remains `std::nullopt` when the option is not specified.

I will do some more thorough testing to confirm.

Reply:
After further testing I came to understand what you meant and made this adjustment as suggested.